src/System.Security.Cryptography.Pkcs/tests/AlgorithmIdentifierTest.cs

```diff
@@ -0,0 +1,112 @@
+            Assert.Equal(0, ai.KeyLength);
+            Assert.Equal(Oids.Rsa, ai.Oid.Value);
+            Assert.Equal(typeof(byte), ai.Parameters.GetType().GetElementType());
+            Assert.Equal(0, ai.Parameters.Length);
+        }
+
+        [Fact]
+        public static void ConstructorTakesNullOid()
+        {
+            AlgorithmIdentifier ai = new AlgorithmIdentifier(null);
+            Assert.Null(ai.Oid);
+            Assert.Equal(0, ai.KeyLength);
+            Assert.Equal(typeof(byte), ai.Parameters.GetType().GetElementType());
+            Assert.Equal(0, ai.Parameters.Length);
+        }
+
+        [Fact]
+        public static void ConstructorTakesOidAndKeyLength()
+        {
+            Oid o = new Oid(Oids.Rsa);
+            AlgorithmIdentifier ai = new AlgorithmIdentifier(o, 128);
+            Assert.Equal(128, ai.KeyLength);
+            Assert.Equal(Oids.Rsa, ai.Oid.Value);
+            Assert.Equal(typeof(byte), ai.Parameters.GetType().GetElementType());
+            Assert.Equal(0, ai.Parameters.Length);
+        }
+
+        [Fact]
+        public static void ConstructorTakesNullOidAndKeyLength()
+        {
+            AlgorithmIdentifier ai = new AlgorithmIdentifier(null, 128);
+            Assert.Null(ai.Oid);
+            Assert.Equal(128, ai.KeyLength);
+            Assert.Equal(typeof(byte), ai.Parameters.GetType().GetElementType());
+            Assert.Equal(0, ai.Parameters.Length);
+        }
+
+        [Fact]
+        public static void ConstructorTakesOidAndNegativeKeyLength()
+        {
+            Oid o = new Oid(Oids.Rsa);
+            AlgorithmIdentifier ai = new AlgorithmIdentifier(o, -1);
+            Assert.Equal(-1, ai.KeyLength);
+            Assert.Equal(Oids.Rsa, ai.Oid.Value);
+            Assert.Equal(typeof(byte), ai.Parameters.GetType().GetElementType());
+            Assert.Equal(0, ai.Parameters.Length);
+        }
+
+        [Fact]
+        public static void KeyLength()
+        {
+            AlgorithmIdentifier ai = new AlgorithmIdentifier
+            {
+                KeyLength = int.MaxValue
+            };
+            Assert.Equal(int.MaxValue, ai.KeyLength);
+            ai.KeyLength = 0;
+            Assert.Equal(0, ai.KeyLength);
+            ai.KeyLength = int.MinValue;
+            Assert.Equal(int.MinValue, ai.KeyLength);
+        }
+
+        [Fact]
+        public static void Oid()
+        {
+            AlgorithmIdentifier ai = new AlgorithmIdentifier
+            {
+                Oid = new Oid(Oids.Rsa)
+            };
+            Assert.Equal(Oids.Rsa, ai.Oid.Value);
+            ai.Oid = null;
+            Assert.Null(ai.Oid);
+        }
+
+        [Fact]
+        public static void Parameters()
+        {
+            AlgorithmIdentifier ai = new AlgorithmIdentifier
+            {
+                Parameters = new byte[2] { 0x05, 0x00 } // ASN.1 NULL
+            };
+            Assert.Equal("05-00", BitConverter.ToString(ai.Parameters));
```

Comment:
For the crypto tests there's a more standard hex form.  Add a using for "Test.Cryptography", then this becomes

```C#
Assert.Equal("0500", ai.Parameters.ByteArrayToHex());
```